### PR TITLE
Add messaging exception documentation

### LINKED ISSUE #15402: Document messaging service errors
**Description**

There's a fixed number of possible, common messaging errors which may need to be handled on every call. This should be documented clearly somewhere (TBD) to ensure we remember handling them.

### PR DESCRIPTION
**Pull request by npepinpe**
## Description

This PR adds very basic messaging documentation, with a focus on exception handling at the messaging service layer. It also adds more concise Javadoc for the possible exception cases.

## Related issues

closes #15402 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Other teams:
If the change impacts another team an issue has been created for this team, explaining what they need to do to support this change.
- [ ] [Operate](https://github.com/camunda/operate/issues)
- [ ] [Tasklist](https://github.com/camunda/tasklist/issues)
- [ ] [Web Modeler](https://github.com/camunda/web-modeler/issues)
- [ ] [Desktop Modeler](https://github.com/camunda/camunda-modeler/issues)
- [ ] [Optimize](https://github.com/camunda/camunda-optimize/issues)

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

File: atomix/cluster/src/main/java/io/atomix/cluster/messaging/MessagingService.java [hunk 1]
```diff
@@ -70,7 +78,23 @@ CompletableFuture<Void> sendAsync(
...
+   *       (e.g. the service is not running)
+   *   <li>{@link java.net.ConnectException} - indicates the recipient is unreachable
+   *   <li>{@link java.util.concurrent.TimeoutException} - indicates no response came back after the
+   *       default timeout
+   *   <li>{@link io.atomix.cluster.messaging.MessagingException.ProtocolException} - indicates the
+   *       recipient failed to parse the incoming message
+   *   <li>{@link io.atomix.cluster.messaging.MessagingException.NoRemoteHandler} - indicates the
+   *       recipient received the message, but was no expecting to
+   *   <li>{@link io.atomix.cluster.messaging.MessagingException.RemoteHandlerFailure} - indicates
+   *       the recipient parsed and expected the message, but it failed unexpectedly to process it
```
Comment (Zelldon) [inline atomix/cluster/src/main/java/io/atomix/cluster/messaging/MessagingService.java:96]: 🔧 I think correct java doc would be to use throws right? 🤔 But anyway it is great you documented it 👍🏼

File: docs/messaging.md [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+  - If the server cannot understand this protocol version, it will write back its own latest
+    protocol version.
+- The client reads whichever version the server wrote back, and if it can, activates this version.
+- If the client cannot, it will close the connection preemptively.
+- If the handshake is successful, _all messages on this channel are expected to be of this version,
+  and the handshake is never performed again._
+
+### Protocol format
+
+> [!Note] Version 2 has been around for years, so it's almost impossible to run into version 1 in
```
Comment (Zelldon) [inline docs/messaging.md:43]:
```suggestion
> [!Note] 
> Version 2 has been around for years, so it's almost impossible to run into version 1 in
```

File: docs/messaging.md [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+  replies, such that this error is returned _only_ for unexpected exceptions, which would normally
+  not be retry-able.
+
+On top of these, you also have an additional exception when using
+the [ClusterCommunicationService](/atomix/cluster/src/main/java/io/atomix/cluster/messaging/ClusterCommunicationService.java),
+which is used specifically to communicate between members of the same cluster via their IDs.
+
+- `io.atomix.cluster.messaging.MessagingException.NoSuchMemberException` - this can occur in two
+  ways. This can be returned immediately at the calling site (so on the client side) if the there is
+  known member with the given ID in the cluster (obtained via the membership protocol). It can also
```
Comment (Zelldon) [inline docs/messaging.md:121]: ❓ I think you mean `no known member` or?

**Review by ChrisKujawa — APPROVED** [review]
Thanks great stuff @npepinpe 🚀

File: atomix/cluster/src/main/java/io/atomix/cluster/messaging/MessagingService.java (hunk 1 above)
Comment (npepinpe) [reply atomix/cluster/src/main/java/io/atomix/cluster/messaging/MessagingService.java:96]: I tried but then later when you build/release the javadoc, it complains that the method doesn't actually throw the exception :sweat_smile:

File: docs/messaging.md (hunk 3 above)
Comment (npepinpe) [reply docs/messaging.md:122]: :see_no_evil: Yes, no known member

**Pull request merged**